src/plugins/minimal_scene/MinimalSceneRhi.hh

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef IGNITION_GUI_PLUGINS_MINIMALSCENE_MINIMALSCENERHI_HH_
+#define IGNITION_GUI_PLUGINS_MINIMALSCENE_MINIMALSCENERHI_HH_
+
+#include "ignition/gui/Plugin.hh"
+#include "ignition/rendering/RenderTypes.hh"
+
+#include <QSGTexture>
+#include <QSize>
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  class IgnCameraTextureRhi
+  {
+    public: virtual ~IgnCameraTextureRhi();
+    public: virtual void Update(rendering::CameraPtr _camera) = 0;
+    public: virtual void TextureId(void* _texturePtr) = 0;
+  };
+
+  /// \brief Ign-rendering renderer.
+  class IgnRenderer;
+  class RenderSync;
+
+  /// \brief Render interface class to handle OpenGL / Metal compatibility
```

Comment:
Class require documentation - if the approach is acceptable then this class and others will be fully documented as per contrib. guidelines.